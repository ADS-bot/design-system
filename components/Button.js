import PropTypes from 'prop-types'
import React from 'react'
import { Button as RButton } from 'rebass/styled-components'

export const Button = props => {
  let variant = 'primary'
  if (props.hollow == true) {
    variant = 'hollow'
  }
  if (props.inverted == true) {
    variant = 'inverted'
  }
  if (props.variant !== undefined) {
    variant = props.variant
  }
  return <RButton
    {...props}
    variant={variant}
    sx={{
      borderWidth: props.border
    }}
  />
}

Button.propTypes = {
  fontSize: PropTypes.number,
  inverted: PropTypes.bool,
  hollow: PropTypes.bool,
  border: PropTypes.number,
  variant: PropTypes.string
}

Button.defaultProps = {
  fontSize: 1,
  border: 2
}

export default Button
