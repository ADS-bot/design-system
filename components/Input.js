import React from 'react'

import { space, width, fontSize, color } from 'styled-system'
import styled from 'styled-components'

export const Input = styled.input`
  $space;
  $width;
  $fontSize;
  $color;

  font-family: inherit;
  font-size: 1rem;
  display: 'block';
  vertical-align: 'middle';

  background-color: transparent;

  border: none;
  border-bottom: 1px solid ${props => props.theme.colors.gray2};
  border-radius: 0;
  margin: 0;
  margin-bottom: 15px;
  padding: 0;

  outline: none;
  width: 100%;
  height: 2rem;

  box-shadow: none;
  box-sizing: content-box;
  transition: all .3s;

  &:focus:not([readonly]) {
    border-bottom: 1px solid ${props => props.theme.colors.base};
    box-shadow: 0 1px 0 0 ${props => props.theme.colors.gray1};
  }
`
export default Input
