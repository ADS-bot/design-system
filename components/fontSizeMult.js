/* global require */
import {
  system
} from 'styled-system'

const {
  defaultFontSizes
} = require('./constants')

export const fontSizeMult = (propName, k) => {
  /* eslint-disable no-console */
  console.error('usage of fontSizeMult will be deprecated')
  /* eslint-enable no-console */

  return system({
    'fontSize': {
      property: propName,
      scale: 'fontSizes',
      transform: n => n * k,
      defaultScale: defaultFontSizes
    }
  })
}

export default fontSizeMult
