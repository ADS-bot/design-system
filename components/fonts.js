import { injectGlobal } from 'styled-components'

import CharterRegular from '../fonts/Charter-Regular.woff'
import CharterBold from '../fonts/Charter-Bold.woff'

import FiraSansLight from '../fonts/FiraSans-Light.woff'
import FiraSansRegular from '../fonts/FiraSans-Regular.woff
import FiraSansBold from '../fonts/FiraSans-Bold.woff'

import SourceCodeProBold from '../fonts/SourceCodePro-Bold.woff'
import SourceCodeProRegular from '../fonts/SourceCodePro-Regular.woff'


injectGlobal`
  @font-face {
    font-family: "Fira Sans";
    src: url('${FiraSansLight}') format('woff');
    font-style: normal;
    font-weight: 300;
  }

  @font-face {
    font-family: "Fira Sans";
    src: url('${FiraSansRegular}') format('woff');
    font-style: normal;
    font-weight: 400;
  }

  @font-face {
    font-family: "Fira Sans";
    src: url('${FiraSansSemiBold}') format('woff');
    font-style: normal;
    font-weight: 700;
  }

  @font-face {
    font-family: "Source Code Pro";
    src: url('${SourceCodeProRegular}') format('woff');
    font-style: normal;
    font-weight: 400;
  }

  @font-face {
    font-family: "Source Code Pro";
    src: url('${SourceCodeProBold}') format('woff');
    font-style: normal;
    font-weight: 700;
  }

  @font-face {
    font-family: "Charter";
    src: url('${CharterRegular}') format('woff');
    font-style: normal;
    font-weight: 300;
  }

  @font-face {
    font-family: "Charter";
    src: url('${CharterBold}') format('woff');
    font-style: normal;
    font-weight: 700;
  }
`
