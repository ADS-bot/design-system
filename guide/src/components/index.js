import  ColorPalette from './ColorPalette'

const GuideComponents = Object.assign({}, { ColorPalette })
export default GuideComponents
