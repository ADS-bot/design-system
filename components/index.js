import {
  Flex,
  Box,
  Link,
} from 'rebass/styled-components'

import { Select, Label } from '@rebass/forms/styled-components'

import { ThemeProvider as Provider } from 'styled-components'

import colors from './theme/colors'
import theme from './theme/rebassTheme'

export { default as colors } from './theme/colors'
export { default as theme } from './theme/rebassTheme'

import components from './components'

// We export these from rebass directly
export {
  Flex,
  Box,
  Link,
} from 'rebass/styled-components'

export { Select, Label } from '@rebass/forms/styled-components'

export const {
  Button,
  ButtonOutline,
  ButtonCircle,
  ButtonTransparent,
  NavLink,
  BlockLink,
  Heading,
  Subhead,
  Text,
  Small,
  Lead,
  Pre,
  Code,
  Samp,
  Blockquote,
  Measure,
  Truncate,
  Input,
  Textarea,
  Checkbox,
  Radio,
  Slider,
  Image,
  Avatar,
  BackgroundImage,
  Container,
  Divider,
  Border,
  Media,
  Card,
  Banner,
  Panel,
  PanelHeader,
  PanelFooter,
  Progress,
  Message,
  Group,
  Toolbar,
  Badge,
  Circle,
  Tabs,
  TabItem,
  DotButton,
  Close,
  Relative,
  Absolute,
  Fixed,
  Sticky,
  Drawer,
  Overlay,
  Carousel,
  ScrollCarousel,
  CarouselSlide,
  Tooltip,
  Switch,
  Arrow,
  Star,
  Embed,
  Donut,
  Row,
  Column,
  RadioGroup,
  RadioButton,
  HeroLead,
  Hero,
  LogoOONIRun,
  InputWithIconButton,
  TwitterShareButton,
  FacebookShareButton,
  Modal,
  BarChart,
  PieChart
} = components

export const OONIComponents = Object.assign({}, components, {
  Provider,
  Box,
  Flex,
  Link,
  Select,
  Label,
  colors,
  theme
})

export default OONIComponents
