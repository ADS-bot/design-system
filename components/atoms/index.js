import Input from './Input'
import Heading from './Heading'
import Text from './Text'
import IconButton from './IconButton'
import { RadioGroup, RadioButton } from './Radio'
import Container from './Container'
import { TwitterShareButton, FacebookShareButton } from './ShareButton'

const atoms = {
  Text,
  Heading,
  IconButton,
  Input,
  RadioGroup,
  RadioButton,
  TwitterShareButton,
  FacebookShareButton,
  Container
}

export default atoms
