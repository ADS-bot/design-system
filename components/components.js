import atoms from './atoms'
import molecules from './molecules'
import organisms from './organisms'

import BarChart from './charts/BarChart'
import PieChart from './charts/PieChart'

/* OONI produced atoms */
import Button from './Button'
export { default as Button } from './Button'
export const Input = atoms.Input
export const RadioButton = atoms.RadioButton
export const RadioGroup = atoms.RadioGroup
export const TwitterShareButton = atoms.TwitterShareButton
export const FacebookShareButton = atoms.FacebookShareButton
export const Heading = atoms.Heading
export const Text = atoms.Text
export const Container = atoms.Container

/* OONI produced molecules */
export const InputWithIconButton = molecules.InputWithIconButton
export const LogoOONIRun = molecules.LogoOONIRun
export const Card = molecules.Card

/* OONI produced organisms */
export const Hero = organisms.Hero
export const HeroLead = organisms.HeroLead
export const Modal = organisms.Modal

/* Here we just export rebass until we implement all of these */
import {
  Absolute,
  Arrow,
  Avatar,
  BackgroundImage,
  Badge,
  Banner,
  BlockLink,
  Blockquote,
  Border,
  ButtonCircle,
  ButtonOutline,
  ButtonTransparent,
  Carousel,
  CarouselSlide,
  Circle,
  Close,
  Code,
  Column,
  Divider,
  Donut,
  DotButton,
  Drawer,
  Embed,
  Fixed,
  Group,
  Image,
  Lead,
  Measure,
  Media,
  Message,
  NavLink,
  Overlay,
  Panel,
  PanelFooter,
  PanelHeader,
  Pre,
  Progress,
  Relative,
  Row,
  Samp,
  ScrollCarousel,
  Small,
  Star,
  Sticky,
  Subhead,
  Switch,
  TabItem,
  Tabs,
  Toolbar,
  Tooltip,
  Truncate
} from 'rebass/styled-components'

export { default as BarChart } from './charts/BarChart'
export { default as PieChart } from './charts/PieChart'

const components = {
  Button,
  Heading,
  Text,
  Input,
  Card,
  RadioButton,
  RadioGroup,
  InputWithIconButton,
  HeroLead,
  Hero,
  LogoOONIRun,
  TwitterShareButton,
  FacebookShareButton,
  Modal,
  BarChart,
  PieChart,
  Absolute,
  Arrow,
  Avatar,
  BackgroundImage,
  Badge,
  Banner,
  BlockLink,
  Blockquote,
  Border,
  ButtonCircle,
  ButtonOutline,
  ButtonTransparent,
  Carousel,
  CarouselSlide,
  Circle,
  Close,
  Code,
  Column,
  Container,
  Divider,
  Donut,
  DotButton,
  Drawer,
  Embed,
  Fixed,
  Group,
  Image,
  Lead,
  Measure,
  Media,
  Message,
  NavLink,
  Overlay,
  Panel,
  PanelFooter,
  PanelHeader,
  Pre,
  Progress,
  Relative,
  Row,
  Samp,
  ScrollCarousel,
  Small,
  Star,
  Sticky,
  Subhead,
  TabItem,
  Tabs,
  Toolbar,
  Tooltip,
  Truncate
}

export default components
