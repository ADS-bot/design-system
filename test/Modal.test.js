/* global describe, test, expect, jest */
import React from 'react'
import { renderJSON, shallowWithTheme } from './index'
import { Modal } from '../components'

describe('Modal', () => {
  test('renders', () => {
    const json = renderJSON(
      <Modal />
    )
    expect(json).toMatchSnapshot()
  })

  test('renders with close button', () => {
    const json = renderJSON(
      <Modal closeButton='right' />
    )
    expect(json).toMatchSnapshot()
  })

  test('close button click handler is called', () => {
    const onCloseClick = jest.fn()
    const wrapper = shallowWithTheme(
      <Modal closeButton='right' onCloseClick={onCloseClick} />
    )
    wrapper.find('Modal__StyledCloseButton').simulate('click')
    expect(onCloseClick).toHaveBeenCalled()
  })

})
