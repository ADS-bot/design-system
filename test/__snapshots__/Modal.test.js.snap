// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: relative;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: fixed;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  max-width: 100vw;
  max-height: 100vh;
  overflow: auto;
  z-index: 1050;
  background-color: #f8f9fa;
  border-radius: radius;
}

<div
  class="c0"
>
  <div
    class="c1"
  />
</div>
`;

exports[`Modal renders with close button 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: relative;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: fixed;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  max-width: 100vw;
  max-height: 100vh;
  overflow: auto;
  z-index: 1050;
  background-color: #f8f9fa;
  border-radius: radius;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: #f8f9fa;
  background-color: #0588cb;
  border: 0;
  border-radius: 4px;
  font-size: 22px;
  font-weight: 700;
  color: #adb5bd;
  background-color: transparent;
  border-radius: none;
  border-width: 1px;
  border-style: solid;
  border-color: #0588cb;
  border: none;
  border-width: 2px;
  font-size: 16px;
}

.c2:hover {
  color: #ced4da;
}

.c2:disabled {
  opacity: 0.5;
}

.c2:active {
  color: #868e96;
}

.c3 {
  position: absolute;
  top: 0;
  height: 28px;
  padding: 0;
  margin: 0;
  left: initial;
  right: 0;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <button
      class="c2 c3"
      font-size="1"
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style="vertical-align: middle;"
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m31.6 10.7l-9.3 9.3 9.3 9.3-2.3 2.3-9.3-9.3-9.3 9.3-2.3-2.3 9.3-9.3-9.3-9.3 2.3-2.3 9.3 9.3 9.3-9.3z"
          />
        </g>
      </svg>
    </button>
  </div>
</div>
`;
